sql/moz-fx-data-shared-prod/org_mozilla_ios_firefox_derived/experiment_search_events_live_v1/init.sql

```diff
@@ -0,0 +1,93 @@
+-- This and the following materialized view need to be kept in sync:
+-- - org_mozilla_fenix_derived.experiment_search_events_live_v1
+-- - org_mozilla_firefox_beta_derived.experiment_search_events_live_v1
+-- - org_mozilla_firefox_derived.experiment_search_events_live_v1
+-- - org_mozilla_ios_firefox_derived.experiment_search_events_live_v1
+-- - org_mozilla_ios_firefoxbeta_derived.experiment_search_events_live_v1
+CREATE MATERIALIZED VIEW
+IF
+  NOT EXISTS `moz-fx-data-shared-prod.org_mozilla_ios_firefox_derived.experiment_search_events_live_v1`
+  OPTIONS
+    (enable_refresh = TRUE, refresh_interval_minutes = 5)
+  AS
+  -- Materialized views do not support sub-queries for UNNESTing fields.
+  -- This limitation prevents summing up individual nested counters.
+  -- As a workaround the nested counters are converted into arrays of counts. The
+  -- conversion is done by a combination of dumping the fields as JSON
+  -- strings, applying regexes and concatenating extracted strings.
+  --
+  -- (1) Dump the nested counter field as JSON string, for example:
+  --     [{"key":"duckduckgo","value":7}, {"key":"google","value":4}]
+  -- (2) Using a regex, extract all values and dump the result array as
+  --     JSON string: ["7", "4"]
+  -- (3) Using regex replace, transform the count numbers in the array
+  --     to an array where the position of the element determines the
+  --     count type [search_ad_clicks, search_with_ads, search_count]:
+  --     "[[7, 0, 0], [4, 0, 0]]"
+  -- (4) Convert the nested array string to an array and concatenate
+  --     nested arrays of ping to a single array and dump as JSON string,
+  --     for example: "[[7, 0, 0], [4, 0, 0], [0, 1, 0], [0, 0, 3]]"
+  --
+  -- (5) Convert the nested array string into an array, unnest and sum
+  --     individual counts based on their array index:
+  --        ad_clicks_count = 11
+  --        search_with_ads_count = 1
+  --        search_count = 3
+  WITH aggregated_arrays AS (
+    SELECT
+      submission_timestamp,
+      experiment.key AS experiment,
+      experiment.value.branch AS branch,
+      -- (4)
+      TO_JSON_STRING(
+        JSON_EXTRACT_ARRAY(
+          -- (3)
+          REGEXP_REPLACE(
+            -- (2)
+            TO_JSON_STRING(
+              REGEXP_EXTRACT_ALL(
+                -- (1)
+                TO_JSON_STRING(metrics.labeled_counter.search_counts),
```

Comment:
For iOS there are fewer search metrics available compared to Fenix or Desktop. `ad_clicks_count` and `search_with_ads_count` will always be 0 until support for those is added.